justfile

```diff
@@ -2,11 +2,26 @@ NAME := "cluster-api-fleet-controller"
 KUBE_VERSION := env_var_or_default('KUBE_VERSION', '1.26.3')
 ORG := "ghcr.io/rancher-sandbox"
 TAG := "dev"
+HOME_DIR := env_var('HOME')
 
 [private]
 default:
     @just --list --unsorted --color=always
 
+# Generates stuff
+generate:
+    just generate-crds
+
+# generates files for CRDS
+generate-crds: install-kopium
+    just _generate-kopium-url {{home_directory()}}/.cargo/bin/kopium "https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/main/config/crd/bases/cluster.x-k8s.io_clusters.yaml" "src/api/capi_cluster.rs"
+    just _generate-kopium-url {{home_directory()}}/.cargo/bin/kopium "https://raw.githubusercontent.com/rancher/fleet/main/charts/fleet-crd/templates/crds.yaml" "src/api/fleet_cluster.rs" "select(.spec.names.singular==\"cluster\")"
+    just _generate-kopium-url {{home_directory()}}/.cargo/bin/kopium "https://raw.githubusercontent.com/rancher/fleet/main/charts/fleet-crd/templates/crds.yaml" "src/api/fleet_clustergroup.rs" "select(.spec.names.singular==\"clustergroup\")"
+
+[private]
+_generate-kopium-url kpath="" source="" dest="" yqexp=".":
+    curl -sSL {{source}} | yq '{{yqexp}}' | {{kpath}} -D Default -f - > {{dest}}
```

Comment:
LGTM, just `yq` is an env requirement here too.

Reply:
Good point, i changed it so it downloads yq: https://github.com/rancher-sandbox/cluster-api-addon-provider-fleet/pull/4/commits/75fcfbf5ff7659b3d33a32755a0b0ca0b6bf26e1

Reply:
Works now, thanks